stac_fastapi/pgstac/core.py

```diff
@@ -54,8 +54,7 @@ async def all_collections(  # noqa: C901
         sortby: Optional[str] = None,
         filter_expr: Optional[str] = None,
         filter_lang: Optional[str] = None,
-        q: Optional[List[str]] = None,
-        **kwargs,
+        **kwargs: Any,
```

Comment:
In this PR we changed and we now forward `kwargs` to `_clean_search_args` method.

Reply:
What happens if `POST /search` with `{"q": 123}` is submitted? Does `q` make it all the way to the DB and raises due to invalid types? There won't be any early API model validation of the parameter?

Reply:
Function `self._clean_search_args` calls are updated with the `**kwargs`, so `q` will now trickle down within these calls as well (which is good), and should be considered (but could be in a separate PR though not to block this one).